code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm

```diff
@@ -260,7 +264,69 @@
 						break
 			if (visible_contents && .)
 				update_icon()
+		if("examine")
+			var/name = params["name"]
+			var/obj/item/target
+			for(var/obj/item/O in src)
+				if(O.name == name)
+					target = O
+					break
+
+			if(!target)
+				return
+
+			//Code copypasted from /mob/verb/examinate
+			if(is_blind(usr))
+				to_chat(usr, "<span class='warning'>You can't see inside [src]!</span>")
+				return
+
+			usr.face_atom(src)
+			var/list/result = target.examine(usr)
+			to_chat(usr, result.Join("\n"))
+			SEND_SIGNAL(usr, COMSIG_MOB_EXAMINATE, target)
+			return
+
+		if("scan")
+			if(!can_scan)
+				return
+
+			var/name = params["name"]
+			var/obj/item/target
+			for(var/obj/item/O in src)
+				if(O.name == name)
+					target = O
+					break
+
+			if(!target)
+				return
+
+			scan(target, usr)
+			return
 
+// -----------------------------
+//  Standard botany smartfridge
+// -----------------------------
+
+/obj/machinery/smartfridge/plants
+	name = "smartfridge"
+	can_scan = TRUE
+	var/obj/item/plant_analyzer/scanner
+
+/obj/machinery/smartfridge/plants/accept_check(obj/item/O)
+	if(istype(O, /obj/item/reagent_containers/food/snacks/grown/) || istype(O, /obj/item/seeds/) || istype(O, /obj/item/grown/))
+		return TRUE
+	return FALSE
+
+/obj/machinery/smartfridge/plants/Initialize()
+	. = ..()
+	scanner = new
```

Comment:
I agree with Fran, scanner should be supplied to it when building instead of appearing out of nowhere

Reply:
I disagree with the notion of "appearing out of nowhere" - the scanner only exists in terms of code, from the point of view of the user, it's just part of the machine. There should be no way to actually access the scanner itself, and the scanner can only be used on things deposited inside the smartfridge of the right type.

Otherwise... Oh well, if you say so.
Should mapped virology smartfridges start out with a virus extrapolator, or should you have to use up one of them to enable this functionality in them?
Should scanners be insertable into any smartfridge, or just into the smartfridges that currently accept them? That is, should you be able to put the viral extrapolator into a chemistry smartfridge, or only into a virology one?

Reply:
the scanner being part of it is like how arcades and cloners have radios in them or how energy guns have cells purely to reduce boilerplate code

Reply:
> purely to reduce boilerplate code

Not exactly true, it wouldn't be feasible to do this without having an actual item in the machine. This is because both the plant analyzer and virus extrapolator use overrides on the *target* that expects to have the analyzer/extrapolator passed as an argument.

Specifically, for the plant analyzer there are overrides of `attackby` that check if they were attacked by a plant analyzer and display the message for the user. The plant analyzer itself actually has *zero* procs overriden/defined

The virus extrapolator calls `extrapolator_act` on the atom being analyzed, which calls back either `scan` or `extract` on the extrapolator, passing a list of diseases that are present on the atom.